components/aws/sagemaker/DataQualityJobDefinition/src/DataQualityJobDefinition_spec.py

```diff
@@ -0,0 +1,147 @@
+    data_quality_baseline_config: Input
+    data_quality_job_input: Input
+    data_quality_job_output_config: Input
+    job_definition_name: Input
+    job_resources: Input
+    network_config: Input
+    role_arn: Input
+    stopping_condition: Input
+    tags: Input
+
+
+@dataclass
+class SageMakerDataQualityJobDefinitionOutputs(SageMakerComponentBaseOutputs):
+    """Defines the set of outputs for the DataQualityJobDefinition component."""
+
+    ack_resource_metadata: Output
+    conditions: Output
+    sagemaker_resource_name: Output
+
+
+class SageMakerDataQualityJobDefinitionSpec(
+    SageMakerComponentSpec[
+        SageMakerDataQualityJobDefinitionInputs,
+        SageMakerDataQualityJobDefinitionOutputs,
+    ]
+):
+    INPUTS: SageMakerDataQualityJobDefinitionInputs = SageMakerDataQualityJobDefinitionInputs(
+        data_quality_app_specification=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="Specifies the container that runs the monitoring job.",
+            required=True,
+        ),
+        data_quality_baseline_config=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="Configures the constraints and baselines for the monitoring job.",
+            required=False,
+        ),
+        data_quality_job_input=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="A list of inputs for the monitoring job. Currently endpoints are supported as monitoring inputs.",
+            required=True,
+        ),
+        data_quality_job_output_config=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="The output configuration for monitoring jobs.",
+            required=True,
+        ),
+        job_definition_name=InputValidator(
+            input_type=str,
+            description="The name for the monitoring job definition.",
+            required=True,
+        ),
+        job_resources=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="Identifies the resources to deploy for a monitoring job.",
+            required=True,
+        ),
+        network_config=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="Specifies networking configuration for the monitoring job.",
+            required=False,
+        ),
+        role_arn=InputValidator(
+            input_type=str,
+            description="The Amazon Resource Name (ARN) of an IAM role that Amazon SageMaker can assume to perform tasks on y",
+            required=True,
+        ),
+        stopping_condition=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="A time limit for how long the monitoring job is allowed to run before stopping.",
+            required=False,
+        ),
+        tags=InputValidator(
```

Comment:
Can you update to the latest code generator version, there is a fix to make this 1 sentence long.